checker/tests/run-pass/iterator.rs

```diff
@@ -71,7 +71,8 @@ pub fn test3() {
     while let Some(_) = it.next() {
         verify!(it.start <= 10);
     }
-    verify!(it.is_empty());
+    //todo: fix this
+    verify!(it.is_empty()); //~ this is unreachable, mark it as such by using the verify_unreachable! macro
```

Comment:
This seems to be fixed.

Reply:
It succeeds sometimes locally, but fails consistently in CI. I need to spend more time on this, but I wan't you to have a look while you are still around.

Reply:
And now CI is flaky too.

Reply:
This reminds me of the fact that test1 and test2 in this file (commented out) work well locally but also fail in CI.

Reply:
It must be a time-out that only happens on smaller CI machines.

Reply:
I investigated this case and there are some invalid loop invariants such as `it.start == 0`. Need some further investigation.

Reply:
https://github.com/facebookexperimental/MIRAI/blob/c705bec00f089494d9d3597439be884be5e0b06f/checker/src/fixed_point_visitor.rs#L266-L270
A possible fix for this is to update the condition above to
``` rust
if joined_val.is_widened() || !val.eq(joined_val) {
    ...
}
```
because widened values should not be consider to be equal to itself across loop iterations. This would lose precision because some widened values may come from previous loops, rather than the current loop.